From 7dadeb621a60a9b4eab10b35c6b242cdaaffea8f Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Sat, 23 Nov 2013 21:16:35 -0600
Subject: [PATCH 162/248] FogBugz #169918: Map the OCRAM in the device tree
 file.

By mapping the OCRAM in the device tree, this enables peripheral(s) to make
use of the OCRAM by a assigning an iram node, and using:

of_get_named_gen_pool(np, "iram", 0);

This patch also enables the generic SRAM driver that will map the OCRAM into
a generic mempool.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
---
 arch/arm/boot/dts/socfpga.dtsi     | 5 +++++
 arch/arm/configs/socfpga_defconfig | 1 +
 drivers/misc/Kconfig               | 9 +++++++++
 3 files changed, 15 insertions(+)

diff --git a/arch/arm/boot/dts/socfpga.dtsi b/arch/arm/boot/dts/socfpga.dtsi
index f16c8c6..96ec845 100644
--- a/arch/arm/boot/dts/socfpga.dtsi
+++ b/arch/arm/boot/dts/socfpga.dtsi
@@ -645,6 +645,11 @@
 
 		};
 
+		ocram: sram@ffff0000 {
+			compatible = "mmio-sram";
+			reg = <0xffff0000 0x10000>;
+		};
+
 		pmu {
 			#address-cells = <1>;
 			#size-cells = <1>;
diff --git a/arch/arm/configs/socfpga_defconfig b/arch/arm/configs/socfpga_defconfig
index 9f119fc..ddf8e02 100644
--- a/arch/arm/configs/socfpga_defconfig
+++ b/arch/arm/configs/socfpga_defconfig
@@ -178,3 +178,4 @@ CONFIG_VLAN_8021Q_GVRP=y
 CONFIG_GARP=y
 CONFIG_IPV6=y
 CONFIG_CONFIGFS_FS=y
+CONFIG_SRAM=y
diff --git a/drivers/misc/Kconfig b/drivers/misc/Kconfig
index ac43959..6a2ce02 100644
--- a/drivers/misc/Kconfig
+++ b/drivers/misc/Kconfig
@@ -506,6 +506,15 @@ config LATTICE_ECP3_CONFIG
 
 	  If unsure, say N.
 
+config SRAM
+	bool "Generic on-chip SRAM driver"
+	depends on HAS_IOMEM
+	select GENERIC_ALLOCATOR
+	help
+	  This driver allows you to declare a memory region to be managed by
+	  the genalloc API. It is supposed to be used for small on-chip SRAM
+	  areas found on many SoCs.
+
 config ALTERA_HWMUTEX
 	tristate "Altera Hardware Mutex"
 	help
-- 
1.9.1

